src/SystemSolar/First.cpp

```diff
@@ -1,74 +1,185 @@
+    }
+
+    constexpr Vector(T x_, T y_)
+    : x{x_}
+    , y{y_}
+    {
+    }
+
+    constexpr Vector& operator+=(Vector const& o)
+    {
+        x = x + o.x;
+        y = y + o.y;
+        return *this;
+    }
+    constexpr Vector& operator-=(Vector const& o)
     {
-        return {x + other.x, y + other.y};
+        x = x - o.x;
+        y = y - o.y;
+        return *this;
     }
 
-    Vector<T> operator-(const Vector<T>& other) const 
+    constexpr Vector operator*(double k) const
+    {
+        return {x * k, y * k};
+    }
+    constexpr Vector operator/(double k) const
     {
-        return {x - other.x, y - other.y};
+        return {x / k, y / k};
     }
 
-    Vector<T> operator*(T other) const 
+    friend constexpr Vector operator*(double k, Vector const& v)
     {
-        return {x * other, y * other};
+        return v * k;
     }
+};
 
-     Vector<T> operator/(T other) const 
+inline double magnitude(Vector<double> const& v)
+{
+    return std::sqrt(v.x * v.x + v.y * v.y);
+}
+inline Vector<double> normalize(Vector<double> const& v)
+{
+    double m = magnitude(v);
+    if (m == 0.0)
     {
-        return {x / other, y / other};
+        return {0.0, 0.0};
     }
+    else
+    {
+        return v / m;
+    }
+}
 
+struct MetersTag
+{
+};
+struct SecondsTag
+{
+};
+struct KilogramsTag
+{
+};
+struct NewtonsTag
+{
+};
+struct MetersPerSecondTag
+{
 };
 
-using Vec = Vector<double>;
-    
-struct Body 
+using Meters = StrongType<double, MetersTag>;
+using Seconds = StrongType<double, SecondsTag>;
+using Kilograms = StrongType<double, KilogramsTag>;
+using Newtons = StrongType<double, NewtonsTag>;
+using MetersPerSecond = StrongType<double, MetersPerSecondTag>;
+
+using PositionVec = Vector<Meters>;
+using VelocityVec = Vector<MetersPerSecond>;
+using ForceVec = Vector<Newtons>;
+using Vec = ForceVec;
+
+struct Body
 {
     std::string name;
-    double mass;       
-    Vec position;  
-    Vec velocity;  
+    Kilograms mass;
+    PositionVec position;
+    VelocityVec velocity;
 };
 
-// Constante gravitacional
-const double G = 6.67430e-11;
+// Constant gravitational
+constexpr double G = 6.67430e-11;
 
-Vec gravityForce(const Body& a, const Body& b)
+Vec gravityForce(Body const& a, Body const& b)
 {
-    //vector diference a to b
-    Vec diff= b.position - a.position;
-
-    //distance
-    double dist = std::sqrt(diff.x * diff.x + diff.y * diff.y);
-
-    // TODO: calcular distancia y normalizar vector
-    Vec unit = diff * (1.0 / dist);
-    // TODO: aplicar fórmula de Newton: F = G * m1 * m2 / r^2
-    double forceMag= G*((a.mass * b.mass)) / (dist * dist);
-    // TODO: devolver fuerza como vector
-    return unit * forceMag;
+    double const aux = 0.0;
+    double const dx = b.position.x.get() - a.position.x.get();
+    double const dy = b.position.y.get() - a.position.y.get();
+
+    Vector<double> const diff{dx, dy};
+    double const dist2 = (diff.x * diff.x) + (diff.y * diff.y);
+    if (dist2 == aux)
```

Comment:
#should write a function to compare equality between floating-point numbers.

I know it's long but [read this](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/) to understand why.